### ISSUE TITLE
Doom compile should prompt to compile pdf-info-epdfinfo-program

### ISSUE THREAD

**Issue by MithicSpirit** [hlissner/doom-emacs#5086]
### 

- [X] I have read the [contributing guide](https://discourse.doomemacs.org/how2contribute).
- [X] I have searched [the issue tracker](https://github.com/hlissner/doom-emacs/issues?q=label%3A%221.+request%22+) for similar requests.
- [X] I have searched [the module list](https://doomemacs.org/docs/modules.org) in case my feature was already implemented.
- [X] I have confirmed that updating Doom won't fulfill my request.

### Describe your request

`doom compile` already prompts to compile `vterm-module` for the `:term vterm` module. This should be extended to also prompt to compile `pdf-info-epdfinfo-program` for the `:tools pdf` module (currently, it's necessary to open a pdf in Doom, compile `pdf-info-epdfinfo-program`, then run `doom compile` if you don't want it to crash).

### Briefly explain its use-case

1. Install/upgrade Doom (or run `doom clean`)
2. Make sure `:tools pdf` is enabled (and run `doom sync` if it wasn't)
3. Run `doom compile`
4. Be prompted to compile `pdf-info-epdfinfo-program` (or maybe give it a nicer name), answer yes
5. ~~Win~~ `doom compile` doesn't crash; opening pdfs in Doom Emacs works

**Comment by hlissner** [comment]
Compiling dependencies is not `doom compile`'s responsibility. Due to the variable ways these dependencies could be set up, and to stay in line with Doom's "Your system, your rules." mantra, Doom will never try to build vterm for you. There may be room to add a "build epdfinfo" step when building the `pdf-tools` package, but that is another issue altogether, and is dependent on #4273.

In any case, this will not happen with `doom compile`.

**Comment by MithicSpirit** [comment]
Huh so if doom is not building/compiling vterm on `doom compile` what does ``Vterm needs `vterm-module' to work.  Compile it now? (y or n)`` do?

**Comment by hlissner** [comment]
> Huh so if doom is not building/compiling vterm on `doom compile` what does `` Vterm needs `vterm-module' to work. Compile it now? (y or n) `` do?

That's a bug. `use-package` eagerly loads packages when its calls are byte-compiled, and vterm [tries to build itself](https://github.com/akermu/emacs-libvterm/blob/master/vterm.el#L134-L140) when it is loaded. That said, I'm not paying much attention to `doom compile` until the CLI rewrite, since its benefits are negligible anyway.

**Comment by MithicSpirit** [comment]
Ah ok, thank you for the explanation.
